s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InstitutionControllerTest.java

```diff
@@ -0,0 +1,172 @@
+    private AppUserRepository appUserRepository;
+
+    @Autowired
+    private InstitutionRepository institutionRepository;
+
+    @Autowired
+    private UserRoleRepository userRoleRepository;
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private TestDbHelper testDbHelper;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private SlugService slugService;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    private AppUser appUser;
+
+    private String test_institution = "Test Institution";
+    private String slugInstitution;
+
+    @BeforeEach
+    public void beforeEach() {
+        testDbHelper.clean();
+        appUser = appUserRepository.save(AppUser.builder()
+                .email(PROFILE_EMAIL)
+                .name("Get")
+                .surname("Profile")
+                .password(passwordEncoder.encode("password"))
+                .enabled(true)
+                .build());
+
+        slugInstitution = slugService.slugify(test_institution);
+        Institution institution = institutionRepository.save(Institution.builder()
+                .name(test_institution)
+                .slug(slugInstitution)
+                .build());
+        Group group = groupRepository.save(Group.builder().name("__default__").slug("default").institution(institution).build());
+
+        UserRole userRole = UserRole.builder().role(AppRole.GROUP_MEMBER).user(appUser).group(group).build();
+        userRoleRepository.save(userRole);
+    }
+
+    @AfterEach
+    public void afterEach() {
+        testDbHelper.clean();
+    }
+
+    @Test
+    public void shouldGetUserInstitution() throws Exception {
+        mockMvc.perform(get(API_PREFIX_V1 + "/institutions")
+                .contentType(MediaType.APPLICATION_JSON)
+                .with(jwtBearerToken(appUser, objectMapper)))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0].name").value(is(equalTo(test_institution))))
+                .andExpect(jsonPath("$[0].slug").value(is(equalTo(slugInstitution))));
+    }
+
+    @Test
+    public void shouldGetEmptySet() throws Exception {
+        AppUser appUser = appUserRepository.save(AppUser.builder()
+                .email("get2@profile.com")
+                .name("Get")
+                .surname("Profile")
+                .password(passwordEncoder.encode("password"))
+                .enabled(true)
+                .build());
+
+        mockMvc.perform(get(API_PREFIX_V1 + "/institutions")
+                .contentType(MediaType.APPLICATION_JSON)
+                .with(jwtBearerToken(appUser, objectMapper)))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.length()", is(equalTo(0))));
+    }
+
+    @Test
+    public void shouldGetAllUsersInstitutions() throws Exception {
+        String test_institution = "Test Institution ZK";
```

Comment:
Why not reuse the institution from @BeforeEach?

Reply:
beforeeach creates institution for each test, here i need two more.